staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -311,6 +384,29 @@ func InstrumentHandlerFunc(verb, group, version, resource, subresource, scope, c
 	}
 }
 
+// cleanContentType binds the contentType (for metrics related purposes) to a
+// bounded set of known/expected content-types. We retain the charset param only if it is
+// RFC compliant (e.g. https://www.iana.org/assignments/character-sets/character-sets.xhtml)
+func cleanContentType(contentType string) string {
```

Comment:
This seems really expensive for a path that thought it necessary to optimize status code conversion with `codeToString`

Would it be sufficient to:
1. identify the small number of content-types the API server expects to serve (with no charset)
2. prefix-compare the incoming content-type case-insensitively with those
3. record an unknown content-type if none match

The content-type metrics seem useful for judging ratio of yaml to json to proto requests, etc, and this would still allow that

Reply:
I can't remember why exactly codeToString was added (was it @smarterclayton who did that?). But I think it might have been because of fmt.Sprintf() which is really expensive. And seems we also have fmt.Sprintf() here, so I agree we may hit the same issues we had 4 years ago (or whenever it was changed).

I agree with Jordan that the current list of 40 supported content types seems strange. I have never heard of many things in this list. And I don't think we will get anything actionable from here. Does it really matter if it was "font/woff" or "font/woff2"? [whatever those really are...]

Reply:
@tallclair WDYT? I'm ambivalent either way.

Reply:
let's bound this to known mime-types

Reply:
We can memoize the function if performance is the concern, then the call to`fmt.Sprintf` will only happen once per unique content-type string.

Do you have a list of the known and acceptable mime-types? I am quite happy to constrain this list.